include/seqan3/io/alignment_file/input.hpp

```diff
@@ -0,0 +1,1323 @@
+ * \brief The type of the ref offset for the seqan3::field::REF_OFFSET; must model std::UnsignedIntegral.
+ */
+/*!\typedef using flag_type
+ * \memberof seqan3::alignment_file_input_traits_concept
+ * \brief The type of the SAM flag value for the seqan3::field::FLAG; must model std::UnsignedIntegral.
+ */
+/*!\typedef using mapq_type
+ * \memberof seqan3::alignment_file_input_traits_concept
+ * \brief The type of the mapping quality value for the seqan3::field::MAPQ; must model std::UnsignedIntegral.
+ */
+/*!\typedef using e_value_type
+ * \memberof seqan3::alignment_file_input_traits_concept
+ * \brief The type of the e-value for the seqan3::field::EVALUE; must model seqan3::Arithmetic.
+ */
+/*!\typedef using bitscore_type
+ * \memberof seqan3::alignment_file_input_traits_concept
+ * \brief The type of the bitscore value for the seqan3::field::BITSCORE; must model seqan3::Arithmetic.
+ */
+    // TODO alignment type ?!
+//!\}
+//!\cond
+template <typename t>
+SEQAN3_CONCEPT alignment_file_input_traits_concept = requires (t v)
+{
+    // field::SEQ
+    requires Alphabet<typename t::sequence_alphabet>;
+    requires Alphabet<typename t::sequence_legal_alphabet>;
+    requires ExplicitlyConvertibleTo<typename t::sequence_legal_alphabet, typename t::sequence_alphabet>;
+    requires sequence_container_concept<typename t::template sequence_container<typename t::sequence_alphabet>>;
+
+    // field::ID
+    requires Alphabet<typename t::id_alphabet>;
+    requires sequence_container_concept<typename t::template id_container<typename t::id_alphabet>>;
+
+    // field::QUAL
+    requires QualityAlphabet<typename t::quality_alphabet>;
+    requires sequence_container_concept<typename t::template quality_container<typename t::quality_alphabet>>;
+
+    // field::REF_SEQ
+    requires Alphabet<typename t::ref_sequence_alphabet>;
+    requires Alphabet<typename t::ref_sequence_legal_alphabet>;
+    requires ExplicitlyConvertibleTo<typename t::ref_sequence_legal_alphabet, typename t::ref_sequence_alphabet>;
+    requires sequence_container_concept<typename t::template ref_sequence_container<typename t::ref_sequence_alphabet>>;
+
+    // field::REF_ID
+    requires Alphabet<typename t::ref_id_alphabet>;
+    requires sequence_container_concept<typename t::template ref_id_container<typename t::ref_id_alphabet>>;
+
+    // field::OFFSET
+    requires std::UnsignedIntegral<typename t::offset_type>;
+
+    // field::REF_OFFSET
+    requires std::UnsignedIntegral<typename t::ref_offset_type>;
+
+    // field::FLAG
+    requires std::UnsignedIntegral<typename t::flag_type>;
+
+    // field::MAPQ
+    requires std::UnsignedIntegral<typename t::mapq_type>;
+
+    // field::EVALUE
+    requires Arithmetic<typename t::e_value_type>;
+
+    // field::BITSCORE
+    requires Arithmetic<typename t::bitscore_type>;
+
+    // field::ALIGNMENT
+    // You can only specify the container type of the query sequence with is either a sequence container type like
```

Comment:
```suggestion
    // You can only specify the container type of the query sequence which is either a sequence container type like
```

Reply:
Should this also be in the details of `quality_container` in the concept documentation above?